src/org/pentaho/reporting/platform/plugin/SimpleReportingComponent.java

```diff
@@ -452,6 +453,15 @@ public void setInputs( final Map<String, Object> inputs ) {
     if ( inputs.containsKey( DASHBOARD_MODE ) ) {
       dashboardMode = "true".equals( String.valueOf( inputs.get( DASHBOARD_MODE ) ) ); //$NON-NLS-1$
     }
+    if ( inputs.containsKey( QUERY_LIMIT ) ) {
+      try {
+        if ( this.getReport() != null && inputs.get( QUERY_LIMIT ) != null ) {
+          this.getReport().setQueryLimit( Integer.parseInt( (String) inputs.get( QUERY_LIMIT ) ) );
```

Comment:
User cant't enter query limit more than system query limit - this check we have on UI. Why we need check existing query limit? If user enter 200 rows, we should set query limit 200 in report.